src/Adapter/Currency/CommandHandler/BulkToggleCurrenciesStatusHandler.php

```diff
@@ -0,0 +1,131 @@
+use PrestaShop\PrestaShop\Core\Domain\Currency\CommandHandler\BulkToggleCurrenciesStatusHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Currency\Exception\CannotDisableDefaultCurrencyException;
+use PrestaShop\PrestaShop\Core\Domain\Currency\Exception\CannotToggleCurrencyException;
+use PrestaShop\PrestaShop\Core\Domain\Currency\Exception\CurrencyException;
+use PrestaShop\PrestaShop\Core\Domain\Currency\Exception\DefaultCurrencyInMultiShopException;
+use Shop;
+
+/**
+ * Toggles multiple currencies status using legacy Currency object model
+ *
+ * @internal
+ */
+final class BulkToggleCurrenciesStatusHandler extends AbstractCurrencyHandler implements BulkToggleCurrenciesStatusHandlerInterface
+{
+    /**
+     * @var int
+     */
+    private $defaultCurrencyId;
+
+    /**
+     * @param int $defaultCurrencyId
+     */
+    public function __construct($defaultCurrencyId)
+    {
+        $this->defaultCurrencyId = (int) $defaultCurrencyId;
+    }
+
+    /**
+     * @param BulkToggleCurrenciesStatusCommand $command
+     *
+     * @throws CurrencyException
+     */
+    public function handle(BulkToggleCurrenciesStatusCommand $command)
+    {
+        foreach ($command->getCurrencyIds() as $currency) {
+            $entity = new Currency((int) $currency->getValue());
+
+            if ($command->getStatus() == $entity->active) {
+                continue;
+            }
+
+            if (0 >= $entity->id) {
+                throw new CurrencyNotFoundException(sprintf('Currency object with id "%s" has not been found for toggling.', $currency->getValue()));
+            }
+
+            if ($entity->active) {
+                $this->assertDefaultCurrencyIsNotBeingDisabled($entity);
+                $this->assertDefaultCurrencyIsNotBeingDisabledFromAnyShop($entity);
+            }
+
+            try {
+                if (false === $entity->toggleStatus()) {
+                    throw new CannotToggleCurrencyException(sprintf('Unable to toggle Currency with id "%s"', $currency->getValue()));
+                }
+            } catch (PrestaShopException $e) {
+                throw new CurrencyException(sprintf('An error occurred when toggling status for Currency object with id "%s"', $currency->getValue()), 0, $e);
+            }
+        }
+    }
+
+    /**
+     * @param Currency $currency
+     *
+     * @throws CannotDisableDefaultCurrencyException
+     */
+    private function assertDefaultCurrencyIsNotBeingDisabled(Currency $currency)
+    {
+        if ((int) $currency->id === $this->defaultCurrencyId) {
+            throw new CannotDisableDefaultCurrencyException(sprintf('Currency with id "%s" is the default currency and cannot be disabled.', $currency->id));
+        }
+    }
+
+    /**
+     * @param Currency $currency
+     *
+     * @throws DefaultCurrencyInMultiShopException
+     */
+    private function assertDefaultCurrencyIsNotBeingDisabledFromAnyShop(Currency $currency)
```

Comment:
it's a duplicate from BulkDeleteCurrenciesHandler, maybe use an abstract method :thinking: